File: src/domain/entities/service_registration.py
"""Service Registration entity for the Service Registry (Req 4.2)."""

from __future__ import annotations

from dataclasses import dataclass

from src.domain.entities.plugin_manifest import PluginManifest
from src.domain.value_objects import ServiceStatus


@dataclass
class ServiceRegistration:
    """A microservice registered with the Admin Panel.

    Persisted in the Service Registry DynamoDB table.
    ``registration_source`` distinguishes seed-loaded entries from
    runtime API registrations.
    """

    service_name: str
    base_url: str
    health_endpoint: str
    manifest_url: str
    manifest: PluginManifest | None
    min_role: str
    status: ServiceStatus
    version: int
    registered_at: str  # ISO 8601
    updated_at: str  # ISO 8601
    registered_by: str
    registration_source: str  # "seed" | "api"

File: admin-shell/src/domain/entities/ServiceRegistration.ts
export type ServiceStatus = "active" | "degraded" | "inactive";
export type RegistrationSource = "seed" | "api";

export interface PluginManifest {
  name: string;
  version: string;
  entryPoint: string;
  routes: RouteDescriptor[];
  navigation: NavigationEntry[];
  stylesheetUrl?: string;
  configSchema?: Record<string, unknown>;
  healthEndpoint?: string;
  requiredPermissions?: string[];
}

export interface RouteDescriptor {
  path: string;
  requiredRoles: string[];
  label: string;
}

export interface NavigationEntry {
  label: string;
  icon: string;
  path: string;
  requiredRoles: string[];
  group?: string;
  order?: number;
}

export interface ServiceRegistration {
  serviceName: string;
  baseUrl: string;
  healthEndpoint: string;
  manifestUrl: string;
  manifest: PluginManifest | null;
  minRole: string;
  status: ServiceStatus;
  version: number;
  registeredAt: string;
  updatedAt: string;
  registeredBy: string;
  registrationSource: RegistrationSource;
}
